component/otelcol/extension/jaeger_remote_sampling/jaeger_remote_sampling.go

```diff
@@ -0,0 +1,129 @@
+	otelconfig "go.opentelemetry.io/collector/config"
+)
+
+const (
+	DefaultHTTPEndpoint   = "0.0.0.0:5778"
+	DefaultGRPCEndpoint   = "0.0.0.0:14250"
+	DefaaultGRPCTransport = "tcp"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "otelcol.extension.jaeger_remote_sampling",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := jaegerremotesampling.NewFactory()
+
+			return extension.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.extension.jaegerremotesampling component.
+type Arguments struct {
+	GRPC *otelcol.GRPCServerArguments `river:"grpc,block,optional"`
+	HTTP *otelcol.HTTPServerArguments `river:"http,block,optional"`
+
+	Source ArgumentsSource `river:"source,block"`
+}
+
+type ArgumentsSource struct {
+	Content        string                       `river:"content,attr,optional"`
+	Remote         *otelcol.GRPCClientArguments `river:"remote,block,optional"`
+	File           string                       `river:"file,attr,optional"`
+	ReloadInterval time.Duration                `river:"reload_interval,attr,optional"`
+}
+
+var (
+	_ extension.Arguments = Arguments{}
+	_ river.Unmarshaler   = (*Arguments)(nil)
+)
+
+// DefaultArguments holds default settings for otelcol.receiver.zipkin.
+var DefaultArguments = Arguments{}
+
+// Convert implements extension.Arguments.
+func (args Arguments) Convert() otelconfig.Extension {
+	return &jaegerremotesampling.Config{
+		HTTPServerSettings: args.HTTP.Convert(),
+		GRPCServerSettings: args.GRPC.Convert(),
+		Source: jaegerremotesampling.Source{
+			Remote:         args.Source.Remote.Convert(),
+			File:           args.Source.File,
+			ReloadInterval: args.Source.ReloadInterval,
+			Contents:       args.Source.Content,
+		},
+	}
+}
+
+// Extensions implements extension.Arguments.
+func (args Arguments) Extensions() map[otelconfig.ComponentID]otelcomponent.Extension {
+	return nil
+}
+
+// Exporters implements extension.Arguments.
+func (args Arguments) Exporters() map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter {
+	return nil
+}
+
+// UnmarshalRiver applies defaults to args before unmarshaling.
+func (a *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*a = DefaultArguments
+
+	type args Arguments
+	err := f((*args)(a))
+	if err != nil {
+		return err
+	}
+
+	// remote config, local file and contents are all mutually exclusive
+	sourcesSet := 0
+	if a.Source.Content != "" {
+		sourcesSet++
+	}
+	if a.Source.File != "" {
+		sourcesSet++
+	}
+	if a.Source.Remote != nil {
+		sourcesSet++
+	}
+
+	if sourcesSet == 0 {
+		return fmt.Errorf("one of contents, file or remote must be configured")
+	}
+	if sourcesSet > 1 {
+		return fmt.Errorf("only one of contents, file or remote can be configured")
+	}
+
+	if a.GRPC == nil && a.HTTP == nil {
+		return fmt.Errorf("http or grpc must be configured to serve the sampling document")
+	}
```

Comment:
These checks are more appropriate on the UnmarshalRiver rules for `ArgumentsSource` instead of on the outer block (it also narrows down the line information if one of these checks fails).